tests/math_builtin_api/modules/test_generator.py

```diff
@@ -181,52 +185,71 @@ def expand_signature(runner, types, signature):
     for arg in signature.arg_types:
         current_types.extend([types[arg]])
 
-    # Itterate over all possible combinations of namespace, base type, dimension and sign flag.
+    # Iterate over all basic types
     # Try to match all function types (return type and argument types).
-    for var_type in runner.var_types:
-        for base_type in runner.base_types:
-            for dim in runner.dimensions:
-                for unsigned in runner.unsigned:
-                    match = False
-                    nomatch = []
-                    index = 0
-                    for ct in current_types:
-                        if (var_type, base_type, dim, unsigned) in list(ct.keys()):
-                            match = True
-                        else:
-                            nomatch.append(index)
-                        index += 1
-                    if match is True:
-                        # Return value and all arguments are of the same type.
-                        if len(nomatch) == 0:
-                            common_type = current_types[0][(
-                                var_type, base_type, dim, unsigned)]
-                            new_sig = sycl_functions.funsig(signature.namespace, common_type.name, signature.name, [
-                                                            common_type.name for i in range(len(signature.arg_types))], signature.pntr_indx[:])
-                            exp_sig.append(new_sig)
+    all_types = sycl_types.create_basic_types()
+    for name in all_types.keys():
+        match = False
+        nomatch = []
+        index = 0
+        for ct in current_types:
+            ct_keys = ct.keys()
+            ct_part_keys = []
+            for t in ct_keys:
+                ct_part_keys.append(t[4:])
+            if (name, ) in ct_part_keys:
```

Comment:
I'm unsure about his, why not just:
```suggestion
            if name in ct_part_keys:
```